xsuite/src/cli/buildReproducibleAction.ts

```diff
@@ -0,0 +1,196 @@
+import { PathLike, existsSync, mkdirSync, readdirSync } from "node:fs";
+import path from "node:path";
+import { Command } from "commander";
+import { log } from "../_stdio";
+import {
+  getGid,
+  getUid,
+  logAndRunCommand,
+  logError,
+  promptUserWithRetry,
+} from "./helpers";
+
+const defaultReproducibleDockerImage =
+  "multiversx/sdk-rust-contract-builder:v6.1.1";
+
+export const registerBuildReproducibleCmd = (cmd: Command) => {
+  cmd
+    .command("build-reproducible")
+    .description("Build contract in a reproducible way.")
+    .option(
+      "--image <IMAGE_TAG>",
+      "Specify the tag of the docker image, that is used to build the contract (e. g. multiversx/sdk-rust-contract-builder:v6.1.0)",
+    )
+    .option(
+      "--dir <DIR>",
+      "Directory in which the command is executed (default: $(PWD))",
+    )
+    .option(
+      "--output-dir <OUTPUT_DIR>",
+      "Specify the directory where the built artifacts will be saved",
+    )
+    .option(
+      "--no-docker-interactive",
+      "Do not use interactive mode for Docker",
+      true,
+    )
+    .option("--no-docker-tty", "Do not allocate a pseudo-TTY for Docker", true)
```

Comment:
What is it for? Is there a case where we would want docker tty?

Reply:
![image](https://github.com/user-attachments/assets/547a3ab6-56c7-477f-8a26-f23b10dfdd66)